connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -248,6 +227,10 @@ private Long ensureLeaderConfig(long maxOffset, WorkerCoordinator coordinator) {
         toRevoke.putAll(computeDuplicatedAssignments(memberConfigs, connectorAssignments, taskAssignments));
         log.debug("Connector and task to revoke assignments (include duplicated assignments): {}", toRevoke);
 
+        List<WorkerLoad> currentWorkerAssignmentWithoutDuplication =
+            removeDuplicated(workerAssignment(memberConfigs, deleted), toRevoke);
+        log.debug("Complete (excluding deletions and remove duplicated assignments) worker assignments: {}", currentWorkerAssignmentWithoutDuplication);
```

Comment:
change (2): compute the current worker assignment excluding deletions and duplicated assignments. If after excluding deletion and duplicated assignments, there are still workers have assignment higher than totalTasksWeHave / totalWorkers, we still need to revoke more tasks.